include/tnt/core/impl/tensor_impl.hpp

```diff
@@ -26,13 +26,29 @@ namespace tnt
 template <typename DataType>
 inline Tensor<DataType>::Tensor() noexcept {}
 
+TEST_CASE_TEMPLATE("Tensor()", T, test_data_types)
+{
+    Tensor<T> t1;
+    REQUIRE(t1.data == AlignedPtr<T>());
+    REQUIRE(t1.shape == Shape());
+}
+
 template <typename DataType>
 inline Tensor<DataType>::Tensor(const Shape& shape)
 {
     this->shape  = shape;
     this->data   = AlignedPtr<DataType>(shape.total());
 }
 
+TEST_CASE_TEMPLATE("Tensor(Shape&)", T, test_data_types)
+{
+    Shape s1;
+    Tensor<T> t1(s1);
+
+    REQUIRE(t1.shape == s1);
+    REQUIRE(t1.data == AlignedPtr<T>(s1.total()));
```

Comment:
I don't think the content of the pointer is guaranteed to match between allocations. It would be better to test that the length of data == s1.total()